From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PickleHik3 <noreply@example.invalid>
Date: Mon, 26 Oct 2026 12:00:00 +0000
Subject: [PATCH 10/16] Compile out Bluetooth: -Dbluetooth=false

No gnome-bluetooth: no manager, tile, status page or status icon.
---
 src/quick-settings.c     | 12 +++++++++++-
 src/shell.c              |  8 ++++++++
 src/ui/quick-settings.ui |  4 ++++
 src/ui/top-panel.ui      |  9 +++++++++
 4 files changed, 32 insertions(+), 1 deletion(-)

diff --git a/src/quick-settings.c b/src/quick-settings.c
index 1d69dfd..3b7b2dd 100644
--- a/src/quick-settings.c
+++ b/src/quick-settings.c
@@ -12,7 +12,9 @@
 
 #include "quick-settings.h"
 
-#include "bt-status-page.h"
+#if PHOSH_HAVE_BLUETOOTH
+# include "bt-status-page.h"
+#endif
 #include "feedback-status-page.h"
 #include "plugin-loader.h"
 #include "quick-setting.h"
@@ -78,6 +80,7 @@ on_wifi_clicked (PhoshQuickSettings *self, PhoshQuickSetting *child)
 }
 
 
+#if PHOSH_HAVE_BLUETOOTH
 static void
 on_bt_clicked (PhoshQuickSettings *self, PhoshQuickSetting *child)
 {
@@ -91,6 +94,7 @@ on_bt_clicked (PhoshQuickSettings *self, PhoshQuickSetting *child)
   enabled = phosh_bt_manager_get_enabled (manager);
   phosh_bt_manager_set_enabled (manager, !enabled);
 }
+#endif
 
 
 static void
@@ -276,9 +280,13 @@ phosh_quick_settings_class_init (PhoshQuickSettingsClass *klass)
   g_type_ensure (PHOSH_TYPE_QUICK_SETTINGS_BOX);
   g_type_ensure (PHOSH_TYPE_QUICK_SETTING);
 
+#if PHOSH_HAVE_BLUETOOTH
   g_type_ensure (PHOSH_TYPE_BT_STATUS_PAGE);
+#endif
   g_type_ensure (PHOSH_TYPE_FEEDBACK_STATUS_PAGE);
+#if PHOSH_HAVE_NETWORKMANAGER
   g_type_ensure (PHOSH_TYPE_WIFI_STATUS_PAGE);
+#endif
 
   gtk_widget_class_set_template_from_resource (widget_class,
                                                "/mobi/phosh/ui/quick-settings.ui");
@@ -287,7 +295,9 @@ phosh_quick_settings_class_init (PhoshQuickSettingsClass *klass)
 
   gtk_widget_class_bind_template_callback (widget_class, on_wwan_clicked);
   gtk_widget_class_bind_template_callback (widget_class, on_wifi_clicked);
+#if PHOSH_HAVE_BLUETOOTH
   gtk_widget_class_bind_template_callback (widget_class, on_bt_clicked);
+#endif
   gtk_widget_class_bind_template_callback (widget_class, on_battery_clicked);
   gtk_widget_class_bind_template_callback (widget_class, on_rotate_clicked);
   gtk_widget_class_bind_template_callback (widget_class, on_rotate_long_pressed);
diff --git a/src/shell.c b/src/shell.c
index aec77ad..d1d6312 100644
--- a/src/shell.c
+++ b/src/shell.c
@@ -964,12 +964,18 @@ static void
 type_setup (void)
 {
   g_type_ensure (PHOSH_TYPE_BATTERY_INFO);
+#if PHOSH_HAVE_BLUETOOTH
   g_type_ensure (PHOSH_TYPE_BT_INFO);
+#endif
+#if PHOSH_HAVE_NETWORKMANAGER
   g_type_ensure (PHOSH_TYPE_CONNECTIVITY_INFO);
+#endif
   g_type_ensure (PHOSH_TYPE_DOCKED_INFO);
   g_type_ensure (PHOSH_TYPE_FEEDBACK_INFO);
   g_type_ensure (PHOSH_TYPE_HKS_INFO);
+#if PHOSH_HAVE_LOCATION
   g_type_ensure (PHOSH_TYPE_LOCATION_INFO);
+#endif
   g_type_ensure (PHOSH_TYPE_DEFAULT_MEDIA_PLAYER);
   g_type_ensure (PHOSH_TYPE_PASSWORD_ENTRY);
   g_type_ensure (PHOSH_TYPE_QUICK_SETTING);
@@ -2102,6 +2108,7 @@ phosh_shell_get_battery_manager (PhoshShell *self)
   return priv->battery_manager;
 }
 
+#if PHOSH_HAVE_BLUETOOTH
 /**
  * phosh_shell_get_bt_manager:
  * @self: The shell singleton
@@ -2124,6 +2131,7 @@ phosh_shell_get_bt_manager (PhoshShell *self)
   g_return_val_if_fail (PHOSH_IS_BT_MANAGER (priv->bt_manager), NULL);
   return priv->bt_manager;
 }
+#endif
 
 /**
  * phosh_shell_get_docked_manager:
diff --git a/src/ui/quick-settings.ui b/src/ui/quick-settings.ui
index 3427915..91f697c 100644
--- a/src/ui/quick-settings.ui
+++ b/src/ui/quick-settings.ui
@@ -28,6 +28,7 @@
             <signal name="clicked" handler="on_wifi_clicked" object="PhoshQuickSettings" swapped="yes"/>
           </object>
         </child>
+        <!-- tl-if bluetooth -->
         <child>
           <object class="PhoshQuickSetting">
             <property name="visible">1</property>
@@ -39,6 +40,7 @@
             <signal name="clicked" handler="on_bt_clicked" object="PhoshQuickSettings" swapped="yes"/>
           </object>
         </child>
+        <!-- tl-endif -->
         <child>
           <object class="PhoshQuickSetting">
             <property name="visible">1</property>
@@ -106,10 +108,12 @@
     <property name="visible">1</property>
     <property name="pixel-size">16</property>
   </object>
+  <!-- tl-if bluetooth -->
   <object class="PhoshBtInfo" id="btinfo">
     <property name="visible">1</property>
     <property name="pixel-size">16</property>
   </object>
+  <!-- tl-endif -->
   <object class="PhoshBatteryInfo" id="batteryinfo">
     <property name="visible">1</property>
     <property name="pixel-size">16</property>
diff --git a/src/ui/top-panel.ui b/src/ui/top-panel.ui
index 34a5bc7..03954e8 100644
--- a/src/ui/top-panel.ui
+++ b/src/ui/top-panel.ui
@@ -237,6 +237,7 @@
                             <property name="child">wifiinfo</property>
                           </object>
                         </child>
+                        <!-- tl-if bluetooth -->
                         <child>
                           <object class="PhoshRevealer">
                             <property name="visible">1</property>
@@ -244,6 +245,8 @@
                             <property name="child">btinfo</property>
                           </object>
                         </child>
+                        <!-- tl-endif -->
+                        <!-- tl-if vpn -->
                         <child>
                           <object class="PhoshRevealer">
                             <property name="visible">1</property>
@@ -411,18 +414,24 @@
     <property name="pixel-size">16</property>
     <property name="margin-end">8</property>
   </object>
+  <!-- tl-if bluetooth -->
   <object class="PhoshBtInfo" id="btinfo">
     <property name="pixel-size">16</property>
     <property name="margin-end">8</property>
   </object>
+  <!-- tl-endif -->
+  <!-- tl-if vpn -->
   <object class="PhoshVpnInfo" id="vpn_info">
     <property name="pixel-size">16</property>
     <property name="margin-end">8</property>
   </object>
+  <!-- tl-endif -->
+  <!-- tl-if networkmanager -->
   <object class="PhoshConnectivityInfo" id="connectivity_info">
     <property name="pixel-size">16</property>
     <property name="margin-end">8</property>
   </object>
+  <!-- tl-endif -->
   <object class="PhoshDockedInfo" id="docked_info">
     <property name="pixel-size">16</property>
     <property name="margin-end">8</property>
